packages/material-react-table/src/menus/MRT_ColumnActionMenu.tsx

```diff
@@ -232,7 +232,7 @@ export const MRT_ColumnActionMenu = ({
             disabled={showColumnFilters && !enableColumnFilterModes}
             divider={enableGrouping || enableHiding}
             key={1}
-            onClick={handleFilterByColumn}
+            onClick={getState().showColumnFilters ? handleOpenFilterModeMenu : handleFilterByColumn}
```

Comment:
small thing. `showColumnFilters` state is already destructured above, so no need to call `getState()` again.

Also, don't know why this build failed